store/src/main/scala/com/improving/app/store/domain/Store.scala

```diff
@@ -147,7 +151,26 @@ object Store {
         }
       case event: StoreInfoEdited =>
         state match {
-          case _: DraftState  => DraftState(event.getInfo, event.getMetaInfo)
+          case _: DraftState =>
```

Comment:
Don't we normally do the validation in the command processing and not the event processing?

Reply:
Here we have `Option[StoreInfo]` on the event and want to save it as `DraftState.info` which itself is an `EditableStoreInfo`. I decided to use the validators because I need to translate empty strings back to `None`, as can be seen in the `EditableStoreInfo` construction below.

I was thinking it might make sense to male `StoreInfoEdited` give back either `StoreInfo` or `EditableStoreInfo`, but decided against it. Now that I think about it that would be possibly cleaner and less weird in terms of how the code looks to do that... I wasn't really sure and did the thing that came first to me.

Reply:
Yes, it will change with IA-168, but I think it needs to be here for now for tests to work

Reply:
The event can only use None if we make the string optional. Hence why I want to replace with OneOf of Editable & regular Info. I can do that here but would prefer to do it as part of moving state objects to protobuf (IA-182). Will add a comment to the ticket about this if you are ok with it

Reply:
Is there a problem with making the string optional? It sounds like that is actually representing what we want

Reply:
In the event we only want to represent the state of the Store. So if in DraftState we return EditableInfo, in other states regular Info. We don't want to make the string optional in Info because it is required for those states.

Reply:
Ok, perhaps just let's not use validators then. I think it's giving the wrong idea about what this code is doing.